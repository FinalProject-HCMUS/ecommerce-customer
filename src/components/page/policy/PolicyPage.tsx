import type React from 'react';
import Breadcrumb from '../../shared/Breadcrumb';
import PolicyContent from './PolicyContent';
import { policyContent } from '../../../data/policy';

const PolicyPage: React.FC = () => {
  return (
    <div className="min-h-screen bg-white text-gray-800">
      {/* Main content */}
      <div className="container mx-auto px-4 pb-12">
         {/* Breadcrumb navigation */}
      <Breadcrumb
        items={[
          { label: 'Home', path: '/' },
          { label: 'Policy', path: '/policy' },
        ]}
      />
        <PolicyContent title="Content about policy" sections={policyContent} />
      </div>
    </div>
  );
};

export default PolicyPage;
