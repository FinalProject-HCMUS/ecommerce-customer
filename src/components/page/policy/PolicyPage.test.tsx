import { render, screen } from '@testing-library/react';
import PolicyPage from './PolicyPage';

// Mock t function
jest.mock('../../../helpers/i18n', () => ({
  t: (key: string) => key,
}));

// Mock Breadcrumb
jest.mock('../../shared/Breadcrumb', () => (props: any) => (
  <nav data-testid="breadcrumb">{JSON.stringify(props.items)}</nav>
));

// Mock PolicyContent
jest.mock('./PolicyContent', () => (props: any) => (
  <div data-testid="policy-content">{props.title}</div>
));

describe('PolicyPage', () => {
  it('renders breadcrumb and policy content', () => {
    render(<PolicyPage />);
    expect(screen.getByTestId('breadcrumb')).toBeInTheDocument();
    expect(screen.getByTestId('breadcrumb')).toHaveTextContent(
      'breadcrumb.home'
    );
    expect(screen.getByTestId('breadcrumb')).toHaveTextContent(
      'breadcrumb.policy'
    );
    expect(screen.getByTestId('policy-content')).toBeInTheDocument();
    expect(screen.getByTestId('policy-content')).toHaveTextContent(
      'lbl.policy'
    );
  });
});
