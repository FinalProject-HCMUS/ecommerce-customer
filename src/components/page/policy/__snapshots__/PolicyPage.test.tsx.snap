// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyPage matches snapshot 1`] = `
<div>
  <div
    class="min-h-screen bg-white text-gray-800"
  >
    <div
      class="px-4 pb-12"
    >
      <div
        data-testid="mock-breadcrumb"
      >
        <span
          data-testid="breadcrumb-item-0"
        >
          Home
           - 
          /
        </span>
        <span
          data-testid="breadcrumb-item-1"
        >
          Policies
           - 
          /policy
        </span>
      </div>
      <div
        data-testid="mock-policy-content"
      >
        Our Policies
      </div>
    </div>
  </div>
</div>
`;
